### PR TITLE
[pulsar-broker] Added support to force deleting tenant

### PR DESCRIPTION
**Pull request by murong00**
### Motivation

Some users expect an option to force deleting tenant for simplicity, it can be useful in some situations.

### Modifications

Add a optional field to force the deletion of all stuffs related to tenant.

### CODE REVIEW

**Comment by BewareMyPower** [comment]
I've fixed the `testIndividualAck` cpp unit test failure in #8141 , please rebase and commit again, thx!

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminTest.java [hunk 1]
```diff
@@ -455,9 +455,11 @@ public void properties() throws Exception {
             assertEquals(e.getResponse().getStatus(), Status.NOT_FOUND.getStatusCode());
         }
 
+        AsyncResponse response = mock(AsyncResponse.class);
+
         // Check deleting non-existing property
         try {
-            properties.deleteTenant("non-existing");
+            properties.deleteTenant(response, "non-existing", false);
```
Comment (codelipenghui) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminTest.java:462]: Is the test related to this change? I think the test should cover there are namespaces and topics under a tenant, then we delete the tenant forcefully should success.

**Review by codelipenghui — CHANGES_REQUESTED** [review]

**Comment by codelipenghui** [comment]
move to 2.8.0 first

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminTest.java (hunk 1 above)
Comment (murong00) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/AdminTest.java:462]: This change adds a force option so the test need pass this option. I can add a test to cover the above situations later.

**Comment by murong00** [comment]
> move to 2.8.0 first

@codelipenghui I can add a test to cover your comments and we expect this feature as soon as possilbe. Can this be introduced in coming 2.7.0?

**Comment by murong00** [comment]
@codelipenghui @jiazhai I have added a test, please help to take a review when available, thanks.

**Comment by murong00** [comment]
Close this and submit a new pull request using metadata-api.

**Pull request closed without merging**